test/gateways/trexle_test.exs

```diff
@@ -54,109 +53,282 @@ defmodule Gringotts.Gateways.TrexleTest do
     description: "For our valued customer, Mr. Potter"
   ]
 
-  describe "purchase" do
-    test "with valid card" do
-      with_mock HTTPoison,
-        request: fn _method, _url, _body, _headers, _options ->
-          MockResponse.test_for_purchase_with_valid_card()
-        end do
-        assert {:ok, _response} = Trexle.purchase(@amount, @valid_card, @opts)
-      end
-    end
-
-    test "with invalid card" do
-      with_mock HTTPoison,
-        request: fn _method, _url, _body, _headers, _options ->
-          MockResponse.test_for_purchase_with_invalid_card()
-        end do
-        assert {:error, response} = Trexle.purchase(@amount, @invalid_card, @opts)
-        assert response.reason == "Your card's expiration year is invalid."
-      end
-    end
-
-    test "with invalid amount" do
-      with_mock HTTPoison,
-        request: fn _method, _url, _body, _headers, _options ->
-          MockResponse.test_for_purchase_with_invalid_amount()
-        end do
-        assert {:error, response} = Trexle.purchase(@bad_amount, @valid_card, @opts)
-        assert response.status_code == 400
-        assert response.reason == "Amount must be at least 50 cents"
-      end
+  @invalid_amount_response ~s/{"error":"Payment failed","detail":"Amount must be at least 50 cents"}/
+  @invalid_card_response ~s/{"error":"Payment failed","detail":"Your card's expiration year is invalid."}/
+  @valid_request_response ~s/{"response":{"token":"charge_3e89c6f073606ac1efe62e76e22dd7885441dc72","success":true,"captured":false}}/
+  @invalid_token_response ~s/{"error":"Capture failed","detail":"invalid token"}/
+  @valid_token_response ~s/{"response":{"token":"#{@valid_token}","success":true,"captured":true,"amount":299,"status_message":"Transaction approved"}}/
+  @invalid_token_response_for_refund ~s/{"error":"Refund failed","detail":"invalid token"}/
+  @unauthorized_access_response ~s/{reason: "Unauthorized access.", message: "Unauthorized access", raw: ""}/
+
+  describe "Store" do
+    setup do
```

Comment:
This is not what I meant... It is not mandatory to keep setup inside `describe` blocks.
You can do:
```elixir
defmodule Gringotts.Gateways.TrexleTest do
  alias ...
  @valid_card %{...}

  setup do # the module-level setup block
    bypass = Bypass.open()
    opts = @opts ++ [test_url: "http://localhost:#{bypass.port}/api/v1/"]
    {:ok, bypass: bypass, opts: opts}
  end

  describe "..." do # no setup in any of the describe blocks
    test "..." do
    end
  end
  ...
end